jep/201/README.adoc

```diff
@@ -230,6 +230,28 @@ The former just uses the configuration-as-code mechanism for initial setup.
 The latter would apply the configuration when updates are detected on file. It could benefit some way to lock down configuration for
 components configured by the configuration-as-code mechanism to be read-only on web UI.
 
+=== Plugin management
+As part of configuring Jenkins Master we support plugin installation and management. Configuration-as-Code plugin itself is out-of-scope, it has to be bundled with Jenkins so it can take control on master. But for all other plugins we want configuration-as-code to offer :
+
+- a high-level user friendly way to define "feature" plugins to be installed. Typically, end-user want to build github pull requests, he doesn't care this one will required `token-macro` for some low-level implementation reasons.
+- a fully reproducible way to setup a master. Relying on `latest` for transitive dependencies is _not_ an acceptable solution. Upgrades should also be tracked as configuration changes.
+- ability to configure infrastructure : private update sites and http proxy to download plugins.
+
+==== Shrink wrap
+`npm` is a typical sample for a dependency manager which has to support comparable requirements : provide a reproducible dependency management in a moving javascript ecosystem. It supports a [`shrinkwrap` file](https://docs.npmjs.com/cli/shrinkwrap) to compute dependency list from a higher level definition, and be able to exactly reproduce this setup later.
```

Comment:
"a typical example of a..."
"provide reproducible dependency management..."
"to compute a dependency list from a higher-level definition..."